### PR TITLE
Add a job to delete a subject from all ephemera folders

### PR DESCRIPTION
**Pull request by hackmastera**
I will probably just run this job from the console for now, but I wanted some tested code
refs #2604

### CODE REVIEW

**Comment by escowles** [comment]
This PR handles the cases where there is a leftover ID, so it's complementary to https://github.com/pulibrary/figgy/pull/2609, right?

**Comment by hackmastera** [comment]
@escowles yes. there are 40-50 instances for the id we ran into this week. If there are others they would probably prohibit a full reindex.

File: app/jobs/clean_deleted_subject_job.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  def perform(subject_id)
+    adapter = Valkyrie::MetadataAdapter.find(:indexing_persister)
+    qs = adapter.query_service
+    folders = qs.find_inverse_references_by(property: :subject, id: subject_id)
+    csp = ChangeSetPersister.new(metadata_adapter: adapter, storage_adapter: Valkyrie.config.storage_adapter)
+    folders.each do |folder|
+      cs = DynamicChangeSet.new(folder).prepopulate!
+      subjects = cs.subject.reject { |id| id.to_s == subject_id }
+      cs.validate(subject: subjects)
+      csp.save(change_set: cs) if cs.valid?
```
Comment (escowles) [inline app/jobs/clean_deleted_subject_job.rb]: Do you want to log an error if it's not valid?

Comment (hackmastera) [reply app/jobs/clean_deleted_subject_job.rb]: sure good idea

**Pull request merged**